crates/xmtp-networking/src/grpc_api_helper.rs

```diff
@@ -0,0 +1,144 @@
+use xmtp_proto::xmtp::message_api::v1;
+
+use tonic::{metadata::MetadataValue, transport::Channel, Request};
+
+// Do a barebones unpaginated Query gRPC request
+// With optional PagingInfo
+pub async fn query(
+    host: String,
+    topic: String,
+    paging_info: Option<v1::PagingInfo>,
+) -> Result<v1::QueryResponse, tonic::Status> {
+    let mut client = v1::message_api_client::MessageApiClient::connect(host)
```

Comment:
At some point we are going to want to optimize this to re-use a long-lived GRPC connection, given how nice and performant it is that GRPC can multiplex a single HTTP/2 connection.

But that's a problem for another day.